dotnet/src/SemanticKernel/Planning/Plan.cs

```diff
@@ -252,18 +257,34 @@ public async Task<Plan> InvokeNextStepAsync(SKContext context)
             #region Update State
 
             // Update state with result
-            this.State.Update(result.Result.Trim());
+            this.State.Update(resultValue);
 
-            // Update state with named outputs (if any)
-            foreach (var item in step.NamedOutputs)
+            // Update Plan Result in State with matching outputs (if any)
+            bool resultAppended = false;
+            foreach (var item in this.Outputs.Intersect(step.Outputs))
             {
-                // ignore the input key
-                if (item.Key.ToUpperInvariant() == "INPUT")
+                if (resultAppended)
```

Comment:
once `resultAppended = true;` all the rest of the iterations will this this and `continue`, right?
Instead, just call `break;`?

But then it looks like resultAppended is always, unconditionally set to true at the end of the first iteration so maybe not a loop at all?